test/generators/suspenders/jobs_generator_test.rb

```diff
@@ -0,0 +1,140 @@
+require "test_helper"
+require "generators/suspenders/jobs_generator"
+
+module Suspenders
+  module Generators
+    class JobsGeneratorTest < Rails::Generators::TestCase
+      include Suspenders::TestHelpers
+
+      tests Suspenders::Generators::JobsGenerator
+      destination Rails.root
+      setup :prepare_destination
+      teardown :restore_destination
+
+      test "adds gems to Gemfile" do
+        expected_output = <<~RUBY
+          gem "sidekiq"
+        RUBY
+
+        run_generator
+
+        assert_file app_root("Gemfile") do |file|
+          assert_match(expected_output, file)
+        end
+      end
+
+      test "installs gems with Bundler" do
+        output = run_generator
+
+        assert_match(/bundle install/, output)
+      end
+
+      test "generator has a description" do
+        description = "Installs sidekiq for background job processing."
+
+        assert_equal description, Suspenders::Generators::JobsGenerator.desc
+      end
+
+      test "configures raising an error when ActiveJob is not found" do
+        expected_configuration = <<~RUBY
+          require "active_job/logging"
+          require "active_job/log_subscriber"
+
+          ActiveSupport::Notifications.unsubscribe("enqueue.active_job")
+
+          module ActiveJob
+            module Logging
+              class EnqueueLogSubscriber < LogSubscriber
+                define_method :enqueue, instance_method(:enqueue)
+              end
+            end
+          end
+
+          ActiveJob::Logging::EnqueueLogSubscriber.attach_to(:active_job)
+        RUBY
+
+        run_generator
+
+        assert_file app_root("config/initializers/active_job.rb") do |file|
+          assert_match(expected_configuration, file)
+        end
+      end
+
+      test "adds ActiveJob configuration to the application file" do
+        config_path = app_root("config/application.rb")
+        application_file = <<~RUBY
+          config.active_job.queue_adapter = :sidekiq
+          config.action_mailer.deliver_later_queue_name = nil
+          config.action_mailbox.queues.routing = nil
+          config.active_storage.queues.analysis = nil
+          config.active_storage.queues.purge = nil
+          config.active_storage.queues.mirror = nil
+        RUBY
+        File.write(config_path, application_file)
+
```

Comment:
I'm guessing you needed to do this because we're backing up the file in the `prepare_destination` step, correct? When we back up a file, we create a blank one. Until now, this hasn't been a problem. However, in order for the `environment` command to work, I think the corresponding file needs to have a `Rails::Application` declaration.

We can fix this by modifying the `backup` method.

```diff
diff --git a/test/test_helper.rb b/test/test_helper.rb
index 4fcaa7e..0495cd1 100644
--- a/test/test_helper.rb
+++ b/test/test_helper.rb
@@ -77,8 +77,7 @@ module Suspenders::TestHelpers
   end
 
   def backup_file(file)
-    FileUtils.mv app_root(file), app_root("#{file}.bak")
-    touch file
+    FileUtils.copy app_root(file), app_root("#{file}.bak")
   end
 
   def restore_file(file)
``` 

This will alleviate us from having to populate the file in the setup step.

```suggestion

```

Reply:
Nice @stevepolitodesign thank you for this! Makes the test much cleaner.